src/ripple/app/ledger/Ledger.cpp

```diff
@@ -624,7 +629,7 @@ Ledger::setup (Config const& config)
 
     try
     {
-        rules_ = Rules(*this);
+        rules_ = Rules(*this, config.features);
```

Comment:
Yes, I'd like to keep them all. Getting rid of the `rules_(config.features)` in the constructors would require adding a default constructor, and that could cause errors if someone adds code that doesn't add the predefined features. Conversely, the `rules_(config.features)` constructor is not equivalent to the assignment above, so we can't remove that.

But most importantly, even if we could get rid of them, the savings isn't worth it. That isn't _that_ expensive.